quickfix To Actually Ignore Top Posts when requested

NotAdrian opened the pull request:
Due to a design change at Instagram relating to their structure when searching for pictures by tags the "top_posts" arrays are not being filled anymore (in the whole program probably). Since they are simply returning 0 elements and the top post elements are now being automatically found through the normal image search there wasn't a big error crisis for the users but the like ratio should have gone down drastically for some people since the top post are now being counted as if they were actual posts which could potentially be liked. This would lead to people starting to raise their base values because there are 9 pictures per tag which are definitely not going to get liked.

The quick fix currently works like this: I'm raising the inserted base Amount by 9 to delete the first nine objects in the Array at the end of the function if skip_top_posts is set to true.

jm-willy commented:
I have checked it and the structure of the html have not changed, the current xpath: "//main/article/div[1]" highlights the first 9 pictures and "//main/article/div[2]" the rest, I have not checked the function though. What do you think is the problem?

NotAdrian commented:
@jm-willy you are right I took another look and the paths are still the same. 
The problem was this line right here and the wrong usage of the here produced value in the function further down in the code:
`        possible_posts = (
            possible_posts if not skip_top_posts else possible_posts - len(top_posts)
        )`
possible_posts was set correctly with this but they only cut the array(/the available link List) shorter without skipping the first nine entries!!! Which lead to this malfunction of still checking the Top Posts.
I have removed this part of the program now because it wasn't integrated right in the ongoing code 
What I added in my first commit fixes this issue in a simple way and I haven't run into an error since I've been using it that way.

breuerfelix commented:
thanks for this fix ! cool that you figured out why this function didnt work :))
